packages/rainbowkit/src/components/ConnectOptions/MobileOptions.tsx

```diff
@@ -40,66 +78,79 @@ function WalletButton({
   const getMobileUri = mobile?.getUri;
   const coolModeRef = useCoolMode(iconUrl);
+  const initialized = useRef(false);
 
   const i18n = useContext(I18nContext);
 
+  const onConnect = useCallback(async () => {
+    if (id === 'walletConnect') onClose?.();
+    connect?.()?.catch(() => {});
+
+    // We need to guard against "onConnecting" callbacks being fired
+    // multiple times since connector instances can be shared between
+    // wallets. Ideally wagmi would let us scope the callback to the
+    // specific "connect" call, but this will work in the meantime.
+    let callbackFired = false;
+
+    onConnecting?.(async () => {
+      if (callbackFired) return;
+      callbackFired = true;
+
+      if (getMobileUri) {
+        const mobileUri = await getMobileUri();
+
+        if (
+          connector.id === 'walletConnect' ||
+          connector.id === 'walletConnectLegacy'
+        ) {
+          // In Web3Modal, an equivalent setWalletConnectDeepLink routine gets called after
+          // successful connection and then the universal provider uses it on requests. We call
+          // it upon onConnecting; this now needs to be called for both v1 and v2 Wagmi connectors.
+          // The `connector` type refers to Wagmi connectors, as opposed to RainbowKit wallet connectors.
+          // https://github.com/WalletConnect/web3modal/blob/27f2b1fa2509130c5548061816c42d4596156e81/packages/core/src/utils/CoreUtil.ts#L72
+          setWalletConnectDeepLink({ mobileUri, name });
+        }
+
+        if (mobileUri.startsWith('http')) {
+          // Workaround for https://github.com/rainbow-me/rainbowkit/issues/524.
+          // Using 'window.open' causes issues on iOS in non-Safari browsers and
+          // WebViews where a blank tab is left behind after connecting.
+          // This is especially bad in some WebView scenarios (e.g. following a
+          // link from Twitter) where the user doesn't have any mechanism for
+          // closing the blank tab.
+          // For whatever reason, links with a target of "_blank" don't suffer
+          // from this problem, and programmatically clicking a detached link
+          // element with the same attributes also avoids the issue.
+          const link = document.createElement('a');
+          link.href = mobileUri;
+          link.target = '_blank';
+          link.rel = 'noreferrer noopener';
+          link.click();
+        } else {
+          window.location.href = mobileUri;
+        }
+      }
+    });
+  }, [connector, connect, getMobileUri, onConnecting, onClose, name, id]);
+
+  // biome-ignore lint/nursery/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    // When using `reactStrictMode: true` in development mode the useEffect hook
+    // will fire twice. We avoid this by using `useRef` logic here. Works for now.
+    if (isCustomConnector && !initialized.current) {
+      onConnect();
+      initialized.current = true;
+    }
+  }, [isCustomConnector]);
```

Comment:
Since `WalletButton` is just a component it can be good to reuse it on our `MobileLinking` component so we make sure we keep the same functionality as it currently is in rainbowkit